src/core/lifetime.d

```diff
@@ -2139,21 +2199,22 @@ private void moveEmplaceImpl(T)(scope ref T target, return scope ref T source)
         // object in order to avoid double freeing and undue aliasing
         static if (hasElaborateDestructor!T || hasElaborateCopyConstructor!T)
         {
-            // If T is nested struct, keep original context pointer
-            static if (__traits(isNested, T))
-                enum sz = T.sizeof - (void*).sizeof;
-            else
-                enum sz = T.sizeof;
-
+            // If there are members that are nested structs, we must take care
+            // not to erase any context pointers, so we might have to recurse
             static if (__traits(isZeroInit, T))
-                () @trusted { memset(&source, 0, sz); }();
+                wipe(source);
             else
-                () @trusted { memcpy(&source, __traits(initSymbol, T).ptr, sz); }();
+                wipe(source, ref () @trusted { return *cast(immutable(T)*) __traits(initSymbol, T).ptr; } ());
         }
     }
     else static if (__traits(isStaticArray, T))
     {
         for (size_t i = 0; i < source.length; ++i)
+            // @@@REVIEW:
+            // why is this move and not moveEmplace???
+            // target is supposed to be uninitialized,
+            // so this will destruct garbage before moving source!
+            // make a spearate fix or fix in this pull?
             move(source[i], target[i]);
```

Comment:
Good catch; I guess a dedicated PR with a little unittest of its own would be nicer.

Reply:
Aight, will do, thanks!